x-pack/plugins/observability_solution/apm/public/components/app/settings/custom_link/create_edit_custom_link_flyout/filters_section.tsx

```diff
@@ -76,49 +76,52 @@ export function FiltersSection({
-            <EuiFlexItem>
-              <EuiSelect
-                data-test-subj={filterId}
-                id={filterId}
-                fullWidth
-                options={selectOptions}
-                value={key}
-                prepend={i18n.translate('xpack.apm.settings.customLink.flyout.filters.prepend', {
-                  defaultMessage: 'Field',
-                })}
-                onChange={(e) =>
-                  // set value to empty string to reset value when new field is selected
-                  onChangeFilter(e.target.value as FilterKey, '', idx)
-                }
-                isInvalid={!isEmpty(value) && (isEmpty(key) || key === DEFAULT_OPTION.value)}
-              />
-            </EuiFlexItem>
-            <EuiFlexItem>
-              <SuggestionsSelect
-                key={key}
-                dataTestSubj={`${key}.value`}
-                fieldName={key}
-                placeholder={i18n.translate(
-                  'xpack.apm.settings.customLink.flyOut.filters.defaultOption.value',
-                  { defaultMessage: 'Value' }
-                )}
-                onChange={(selectedValue) => onChangeFilter(key, selectedValue as string, idx)}
-                defaultValue={value}
-                isInvalid={!isEmpty(key) && isEmpty(value)}
-                start={moment().subtract(24, 'h').toISOString()}
-                end={moment().toISOString()}
-              />
-            </EuiFlexItem>
-            <EuiFlexItem grow={false}>
-              <EuiButtonEmpty
-                data-test-subj="apmCustomLinkFiltersSectionButton"
-                iconType="trash"
-                onClick={() => onRemoveFilter(idx)}
-                disabled={!value && !key && filters.length === 1}
-              />
-            </EuiFlexItem>
-          </EuiFlexGroup>
+          <React.Fragment key={filterId}>
+            <EuiFlexGroup gutterSize="s" alignItems="center">
+              <EuiFlexItem>
+                <EuiSelect
+                  data-test-subj={filterId}
+                  id={filterId}
+                  fullWidth
+                  options={selectOptions}
+                  value={key}
+                  prepend={i18n.translate('xpack.apm.settings.customLink.flyout.filters.prepend', {
+                    defaultMessage: 'Field',
+                  })}
+                  onChange={(e) =>
+                    // set value to empty string to reset value when new field is selected
+                    onChangeFilter(e.target.value as FilterKey, '', idx)
+                  }
+                  isInvalid={!isEmpty(value) && (isEmpty(key) || key === DEFAULT_OPTION.value)}
+                />
+              </EuiFlexItem>
+              <EuiFlexItem>
+                <SuggestionsSelect
+                  key={key}
+                  dataTestSubj={`${key}.value`}
+                  fieldName={key !== '' ? key : DEFAULT_OPTION.value}
```

Comment:
This fixes the issue but passing the value `DEFAULT` will result in an unnecessary request every time the flyout is opened or whenever the user adds another filter and then another request with the actual field 

What do you think of handling the empty state where we fetch the data? Inside the `SuggestionsSelect`

Reply:
Out of scope of this PR but I noticed that we send request every single time we add a new filter for ALL the filters :(

Reply:
> This fixes the issue but passing the value `DEFAULT` will result in an unnecessary request every time the flyout is opened or whenever the user adds another filter and then another request when the user select the actual field
> 
> What do you think of handling the empty state where we fetch the data? Inside the `SuggestionsSelect`

Passing the value `DEFAULT` is not adding new requests, there were already lots of unnecessary ones, I noticed that issue during development. However, to avoid expanding the scope of the bug and to deliver a user-visible solution, I focused on preventing the error rather than altering the existing logic.

There's already another [bug](https://github.com/elastic/kibana/issues/189987) discovered while working on this but it's not related to the multiple requests. They're probably happening because of how the rendering is handled.

I could create an "improvement" issue (not a bug), but I am not sure how I should label it as it's not really a bug but a performance enhancement. Could you please help me on that @kpatticha?

Reply:
> Out of scope of this PR but I noticed that we send request every single time we add a new filter for ALL the filters :(

And every time a filter value is changed!

Reply:
> What do you think of handling the empty state where we fetch the data? Inside the `SuggestionsSelect`

Thanks for the proposal!

I already considered preventing the request and returning an empty list if the required parameter was missing. However, since `SuggestionsSelect` is also used in other areas, I was concerned that this change might impact error handling elsewhere.

What’s your perspective on this?

Reply:
> I could create an "improvement" issue (not a bug), but I am not sure how I should label it as it's not really a bug but a performance enhancement. Could you please help me on that @kpatticha?

We have the label https://github.com/elastic/kibana/labels/apm%3Aperformance 

> Passing the value DEFAULT is not adding new requests, there were already lots of unnecessary ones

Agree there are already lots of unnecessary ones when we render the filters let's make sure we don't add more

Reply:
> I already considered preventing the request and returning an empty list if the required parameter was missing. However, since SuggestionsSelect is also used in other areas, I was concerned that this change might impact error handling elsewhere.

The root cause of the bug seems to be these lines

https://github.com/elastic/kibana/blob/8.15/x-pack/plugins/observability_solution/apm/public/components/shared/suggestions_select/index.tsx#L53-L69

if we use `SuggestionsSelect` in another place and passing empty string for `fieldName`  we will hit that problem again.

Reply:
Yes it is.

I wasn't thinking on extending the bug scope to check on other areas using `SuggestionsSelect` that may have the same issues.

I'd be glad to tackle the issue from the root in this same PR, as long as we're okay with the fact that it might take a bit more time.

Reply:
After some discussion and deep diving w/@kpatticha (thank you!) we discovered the backend part was not dealing properly with a `wildcard` attribute sent to the query and that was causing internally the request error to Elasticsearch API.

I'll close this PR and open a new one with this new better approach that will solve the problem from the root of it.